### PR TITLE
Fix Enumerable#isAny second param type

### PR DESCRIPTION
**Pull request by theroncross**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://emberjs.com/api/ember/3.0/classes/Ember.NativeArray/methods/isAny?anchor=isAny>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by andy-ms** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

**Review by dwickern — CHANGES_REQUESTED** [review]
Are you sure this is right? The linked documentation even says the second argument defaults to `true`

**Comment by theroncross** [comment]
The true is surrounded by backticks, but the description says:
> key String the property to test
value **String** optional value to test against. Defaults to `true`

Are the docs wrong?

**Comment by dwickern** [comment]
I think the docs are wrong
```js
Ember.A([
    { n: 1, foo: false },
    { n: 2, foo: false }
]).isAny('foo')
// -> false

Ember.A([
    { n: 1, foo: false },
    { n: 2, foo: false }
]).isAny('foo', false)
// -> true
```

**Comment by theroncross** [comment]
Seems fine with either: https://ember-twiddle.com/?openFiles=controllers.application.js%2C

**Comment by dwickern** [comment]
Ah, ok it should actually be `any`
```js
Ember.A([
    { n: 1, foo: false },
    { n: 2, foo: false }
]).isAny('n', 2)
// -> true

Ember.A([
    { n: 1, foo: false },
    { n: 2, foo: false }
]).isAny('n', 3)
// -> false
```

**Comment by theroncross** [comment]
Thanks for calling me on it. I'll update.

File: types/ember/index.d.ts
Comment (dwickern) [inline types/ember/index.d.ts]: Is the second argument here also optional?

Comment (dwickern) [reply types/ember/index.d.ts]: Yes... yes it is https://github.com/DefinitelyTyped/DefinitelyTyped/pull/23772

**Pull request merged**
